test/features/errors/notfound.js

```diff
@@ -66,4 +68,99 @@ describe('404 handling', function() {
             assert.contentType(e, 'application/problem+json');
         });
     });
+    it('should return 404 on deleted revision', function() {
+        return preq.get({
+            uri: server.config.bucketURL + '/revision/668588412'
+        })
+        .then(function() {
+            throw new Error('404 should be returned')
+        })
+        .catch(function(e) {
+            assert.deepEqual(e.status, 404);
+            assert.contentType(e, 'application/problem+json');
+        });
+    });
+    it('should set page_deleted on deleted page', function() {
+        var apiURI = server.config
+            .conf.templates['wmf-sys-1.0.0']
+            .paths['/{module:action}']['x-modules'][0].options.apiURI;
+        var title = 'TestingTitle';
+        var revision = 12345;
+
+        nock.enableNetConnect();
+        var api = nock(apiURI)
+            // The first request should return a page so that we store it.
+        .post('')
+        .reply(200, {
+            'batchcomplete': '',
+            'query': {
+                'pages': {
+                    '11089416': {
+                        'pageid': 11089416,
+                        'ns': 0,
+                        'title': title,
+                        'contentmodel': 'wikitext',
+                        'pagelanguage': 'en',
+                        'touched': '2015-05-22T08:49:39Z',
+                        'lastrevid': 653508365,
+                        'length': 2941,
+                        'revisions': [{
+                            'revid': revision,
+                            'user': 'Chuck Norris',
+                            'userid': 3606755,
+                            'timestamp': '2015-03-25T20:29:50Z',
+                            'size': 2941,
+                            'sha1': 'c47571122e00f28402d2a1b75cff77a22e7bfecd',
+                            'contentmodel': 'wikitext',
+                            'comment': 'Test',
+                            'tags': []
+                        }]
+                    }
+                }
+            }
+        })
+        // Other requests return nothing as if the page is deleted.
+        .post('')
+        .reply(200, {})
```

Comment:
This is not what the MW API returns. The 200 code is returned by the MW API, but not an empty response. Here's the body that it returns:

```javascript
{
  "batchcomplete": "",
  "query": {
    "badrevids": {
      "XXXXXX": {
        "revid": XXXXXX
      }
    }
  }
}
```